module/zfs/arc.c

```diff
@@ -2601,9 +2603,12 @@ arc_space_consume(uint64_t space, arc_space_type_t type)
 	case ARC_SPACE_L2HDRS:
 		aggsum_add(&astat_l2_hdr_size, space);
 		break;
+	case ARC_SPACE_ABD_CHUNK_WASTE:
+		aggsum_add(&astat_abd_chunk_waste_size, space);
+		break;
 	}
 
-	if (type != ARC_SPACE_DATA)
+	if (type != ARC_SPACE_DATA && type != ARC_SPACE_ABD_CHUNK_WASTE)
 		aggsum_add(&arc_meta_used, space);
```

Comment:
Given that the ARC makes eviction decisions based on the metadata used, I think we should increment this if the waste is as a result of a metadata ABD.

Reply:
I see, you'd like `arc_meta_min`/`arc_meta_limit` to apply to the chunk waste of metadata abd's.  I think we could do that based on `ABD_FLAG_META`.

I'm slightly concerned that `arc_evict_meta()` (& friends) would evict more than we intended, since we'd be telling it to evict an amount that includes the chunk waste, but it doesn't know about that, so it will evict that amount of `abd_size` (i.e. not including the chunk waste).

Reply:
Yeah, that was what I was thinking but as I've given it more thought I think it makes sense to just avoid adding more metadata-specific code paths (especially since long-term this could go away). I'm good with this as-is.